packages/components/src/components/brandshape/brandshape.ts

```diff
@@ -0,0 +1,155 @@
+import { createRef, ref } from 'lit/directives/ref.js';
+import { css } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import { debounce } from 'throttle-debounce';
+import { html } from 'lit/static-html.js';
+import componentStyles from '../../styles/component.styles';
+import cx from 'classix';
+import SolidElement from '../../internal/solid-element';
+import type { Ref } from 'lit/directives/ref.js';
+import type { TemplateResult } from 'lit-html';
+
+/**
+ * @summary The Brandshape highlights a piece of content.
+ * @documentation https://solid.union-investment.com/[storybook-brandshape]/brandshape
+ * @status
+ * @since
+ *
+ * @slot - The content inside the brandshape.
+ *
+ * @csspart base - The component's base wrapper.
+ * @csspart content-center - Center content wrapper.
+ * @csspart content-top - Top content wrapper.
+ *
+ * @cssproperty --brandshape-opacity - The opacity of the brandshape.
+ */
+@customElement('sd-brandshape')
+export default class SdBrandshape extends SolidElement {
+  /** The brandshape's theme variant. */
+  @property({ type: String }) variant: 'neutral-100' | 'primary' | 'white' = 'primary';
+
+  /** Defines the form of the brandshape  */
+  @property({ type: String }) form: 'full' | 'top' | 'topCenter' | 'centerBottom' = 'full';
```

Comment:
Question: Wouldn't it be possible to do this with either CSS parts or CSS properties?

```
sd-brandshape{
     --top: none;
     --middle: none;
}
```

```
sd-brandshape::part(top){
display: none;
}
```

This could make it possible to maybe even remove a bunch of JS, wouldn't it?

Furthermore:
Let's call it top, middle, bottom, as center is usually used for horizontal alignment and middle for vertical alignment.

If you have to go the JS route, at least I would convert this thing to an array, that people can write `top,middle` or `top` and `top,middle,bottom` is the default. See https://lit.dev/docs/components/properties/#conversion-type

Your solution might work for CMS, but we don't know what happens in future and we don't want to add a bunch of properties.

Reply:
mhhh I don't think it would reduce the amount of js that much since most of the js is just rendering template which you would need with css aswell. Also this means that even thought you only selected top variant, the whole brandshape would rendered in the dom.

Top, middle, bottom sounds good.

I like the array solution, this also matches my current rendering approach.